src/Illuminate/Support/Str.php

```diff
@@ -636,6 +636,27 @@ public static function limit($value, $limit = 100, $end = '...')
         return rtrim(mb_strimwidth($value, 0, $limit, '', 'UTF-8')).$end;
     }
 
+    /**
+     * Limit the number of characters in a string but to the closest word.
+     *
+     * @param string $value
+     * @param int $limit
+     * @param string $end
+     * @return string
+     */
+    public static function limitTillWord($value, $limit = 100, $end = '...')
+    {
+        if (mb_strwidth($value, 'UTF-8') <= $limit) {
+            return $value;
+        }
+
+        $str = trim(preg_replace('/[\n\r]+/', ' ', strip_tags($value)));
+        $wordWrap = wordwrap($str, $limit);
+        $sentences = explode("\n", $wordWrap);
+
+        return preg_replace('/\W$/', '', $sentences[0]) . $end;
```

Comment:
This is a proposed alternative to this as well which I believe would handle multibyte strings.

```suggestion
        if (mb_strwidth($value, 'UTF-8') <= $limit) {
            return $value;
        }

        $value = trim(preg_replace('/[\n\r]+/', ' ', strip_tags($value)));
        $trimmed = rtrim(mb_strimwidth($value, 0, $limit, '', 'UTF-8'));

        if (mb_substr($value, $limit, 1, 'UTF-8') === ' ') {
            return $trimmed.$end;
        }

        return preg_replace("/(.*)\s.*/", '$1', $trimmed).$end;
```